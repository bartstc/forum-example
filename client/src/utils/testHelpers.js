import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
};