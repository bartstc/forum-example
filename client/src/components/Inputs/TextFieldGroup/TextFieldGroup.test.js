import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextFieldGroup from './TextFieldGroup';
import { checkProps } from '../../../utils/testHelpers';

describe('TextAreaFieldGroup', () => {

  let wrapper;
  const props = {
    name: 'string',
    placeholder: 'string',
    value: 'string',
    label: 'string',
    error: 'string',
    info: 'string',
    type: 'string',
    onChange: jest.fn(),
    disabled: 'string'
  };

  beforeEach(() => {
    wrapper = shallow(<TextFieldGroup {...props} />);
  });

  it('renders TextFieldGroup without errors', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should NOT throw a warning', () => {
    const propsError = checkProps(TextFieldGroup, props);
    expect(propsError).toBeUndefined();
  });

});

