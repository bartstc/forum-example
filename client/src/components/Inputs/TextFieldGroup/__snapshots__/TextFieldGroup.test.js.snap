// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaFieldGroup renders TextFieldGroup without errors 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control form-control-lg is-invalid"
    disabled="string"
    name="string"
    onChange={[MockFunction]}
    placeholder="string"
    type="string"
    value="string"
  />
  <small
    className="form-text text-muted"
  >
    string
  </small>
  <div
    className="invalid-feedback"
  >
    string
  </div>
</div>
`;
